src/oneD/refine.cpp

```
@@ -12,7 +12,7 @@ namespace Cantera
 {
 Refiner::Refiner(Domain1D& domain) :
     m_ratio(10.0), m_slope(0.8), m_curve(0.8), m_prune(-0.001),
-    m_min_range(0.01), m_domain(&domain), m_npmax(3000),
+    m_min_range(0.01), m_domain(&domain), m_npmax(300),
```

3000 may be somewhat excessive, but I think 300 is way too low, depending on the refining tolerances that you are using. I would be annoyed if I spent a long time waiting for a solution only to have it stop because it reached this arbitrary limit.

---

You're right, I think I was just annoyed that I was waiting for simulations to finish that took forever because they never converged and hit the 3000 limit, so I was thinking a "fail early" limit would be better. Would 500 or 1000 be more reasonable?

---

I should note, though, that the default argument for `Sim1D::setMaxGridPoints` sets the max to 300: https://github.com/Cantera/cantera/blob/master/include/cantera/oneD/Sim1D.h#L136

---

I think 1000 would be fine. I don't think it makes any sense for for that argument of `Sim1D::setMaxGridPoints` to have a default value.